src/protocolsupport/protocol/typeremapper/itemstack/ItemStackRemapper.java

```diff
@@ -31,16 +30,17 @@ public static NetworkItemStack remapToClient(ProtocolVersion version, String loc
 
 	public static NetworkItemStack remapFromClient(ProtocolVersion version, String locale, NetworkItemStack itemstack) {
 		if (version == ProtocolVersion.MINECRAFT_PE) {
-			int peCombinedId = PEItems.getModernIdByPEIdData(itemstack.getTypeId(), itemstack.getLegacyData());
-			itemstack.setTypeId(PreFlatteningItemIdData.getIdFromLegacyCombinedId(peCombinedId));
-			itemstack.setLegacyData(PreFlatteningItemIdData.getDataFromLegacyCombinedId(peCombinedId));
+			int modernId = PEItems.getModernIdByPEIdData(itemstack.getTypeId(), itemstack.getLegacyData());
+			itemstack.setTypeId(modernId);
 		} else if (version.isBefore(ProtocolVersion.MINECRAFT_1_13)) {
 			int legacyCombinedId = PreFlatteningItemIdData.formLegacyCombinedId(itemstack.getTypeId(), itemstack.getLegacyData());
 			itemstack.setTypeId(PreFlatteningItemIdData.getModernIdByLegacyCombinedId(legacyCombinedId));
 		} else {
 			itemstack.setTypeId(FlatteningItemId.REGISTRY_FROM_CLIENT.getTable(version).getRemap(itemstack.getTypeId()));
 		}
-		return ItemStackComplexRemapperRegistry.remapFromClient(version, locale, itemstack);
+		itemstack = ItemStackComplexRemapperRegistry.remapFromClient(version, locale, itemstack);
```

Comment:
why not just return it?

Reply:
My guess? He made it like that so he could added a debug log there, and then he removed the log without inlining the return 😛

Reply:
ahh @MrPowerGamerBR nailed it lol